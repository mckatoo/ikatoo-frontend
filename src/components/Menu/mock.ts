import { SideMenuProps } from '.'

export const mockedMenu: SideMenuProps = {
  links: [
    {
      label: 'Sobre',
      to: 'about'
    },
    {
      label: 'Habilidades',
      to: 'skills'
    },
    {
      label: 'Projetos',
      to: 'projects'
    },
    {
      label: 'Contato',
      to: 'contact'
    }
  ],
  social: [
    {
      name: 'github',
      url: 'https://github.com/mckatoo',
      url_icon: '/images/github.svg'
    },
    {
      name: 'youtube',
      url: 'https://youtube.com/mckatoo',
      url_icon: '/images/youtube.svg'
    },
    {
      name: 'linkedin',
      url: 'https://linkedin.com/in/mckatoo',
      url_icon: '/images/linkedin.svg'
    }
  ]
}
