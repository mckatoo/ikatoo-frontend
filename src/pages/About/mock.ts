import { AboutProps } from '.'

export const about: AboutProps = {
  skills: [
    'javascript',
    'dotnet',
    'nodedotjs',
    'html5',
    'css3',
    'typescript',
    'jest',
    'json',
    'git',
    'bootstrap',
    'react',
    'mysql',
    'npm',
    'php',
    'java'
  ],
  title: 'Ol√°. Bem vindo‚ùó',
  description:
    '<p>Me chamo Milton Carlos Katoo, moro em Itapira, interior de S√£o Paulo/Brasil. Pai de uma princesa e filho de excelente cozinheira Italiana e um saldoso Japon√™s faz tudo, sou um desenvolvedor full-stack que ama programa√ß√£o e desenvolvimento de software afim de melhorar a vida das pessoas.</p><p>Pessoa bem organizada, solucionador de problemas, funcion√°rio independente com alta aten√ß√£o aos detalhes.F√£ de animes, mangas, games, s√©ries de TV e filmes. Uma pessoa de fam√≠lia e pai de uma princesa.</p><p>Interessado em todo o espectro de programa√ß√£o e trabalhar em projetos ambiciosos com pessoas positivas.</p><a class="text-mck_aqua underline underline-offset-8" href="https://ikatoo.com.br/contact/" rel="contact"><span>üéâ</span>Vamos fazer algo especial.</a><span>üòÑ</span>'
}
