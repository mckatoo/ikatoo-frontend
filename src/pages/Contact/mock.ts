export default {
  title: 'Entre em contato ğŸ˜„',
  description: `Estou interessado em oportunidades CLT, mas nÃ£o descarto nenhum tipo de trabalho desde que me traga oportunidades de aprendizado. No entanto, se vocÃª tiver outra solicitaÃ§Ã£o ou pergunta, nÃ£o hesite em enviar um email. <a href="mailto:mckatoo@gmail.com?Subject=Contato%20pelo%20site">mckatoo@gmail.com</a>`,
  localization: {
    lat: -22.428850083857423,
    lng: -46.830700405308185
  }
}
