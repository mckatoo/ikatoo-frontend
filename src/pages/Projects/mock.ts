import { ProjectProps } from '.'

export const mockProjects: ProjectProps[] = [
  {
    snapshot: '/images/snap-calm.png',
    description: {
      title: 'Calm Organizador de Criptomoedas',
      subTitle: 'Last update: 2022 - 03',
      content: 'Personal project for learn nextjs.'
    },
    githubLink: 'https://github.com/mckatoo/calm'
  },
  {
    snapshot: '/images/ikatoo.png',
    description: {
      title: 'iKatoo - Site pessoal',
      subTitle: 'Last update: 2022 - 08',
      content: 'Personal Web Site.'
    },
    githubLink: 'https://github.com/mckatoo/ikatoo-frontend'
  },
  {
    snapshot: '/images/snap-calm.png',
    description: {
      title: 'Calm Organizador de Criptomoedas',
      subTitle: 'Last update: 2022 - 03',
      content: 'Personal project for learn nextjs.'
    },
    githubLink: 'https://github.com/mckatoo/calm'
  },
  {
    snapshot: '/images/ikatoo.png',
    description: {
      title: 'iKatoo - Site pessoal',
      subTitle: 'Last update: 2022 - 08',
      content: 'Personal Web Site.'
    },
    githubLink: 'https://github.com/mckatoo/ikatoo-frontend'
  },
  {
    snapshot: '/images/snap-calm.png',
    description: {
      title: 'Calm Organizador de Criptomoedas',
      subTitle: 'Last update: 2022 - 03',
      content: 'Personal project for learn nextjs.'
    },
    githubLink: 'https://github.com/mckatoo/calm'
  },
  {
    snapshot: '/images/ikatoo.png',
    description: {
      title: 'iKatoo - Site pessoal',
      subTitle: 'Last update: 2022 - 08',
      content: 'Personal Web Site.'
    },
    githubLink: 'https://github.com/mckatoo/ikatoo-frontend'
  },
  {
    snapshot: '/images/snap-calm.png',
    description: {
      title: 'Calm Organizador de Criptomoedas',
      subTitle: 'Last update: 2022 - 03',
      content: 'Personal project for learn nextjs.'
    },
    githubLink: 'https://github.com/mckatoo/calm'
  },
  {
    snapshot: '/images/ikatoo.png',
    description: {
      title: 'iKatoo - Site pessoal',
      subTitle: 'Last update: 2022 - 08',
      content: 'Personal Web Site.'
    },
    githubLink: 'https://github.com/mckatoo/ikatoo-frontend'
  },
  {
    snapshot: '/images/snap-calm.png',
    description: {
      title: 'Calm Organizador de Criptomoedas',
      subTitle: 'Last update: 2022 - 03',
      content: 'Personal project for learn nextjs.'
    },
    githubLink: 'https://github.com/mckatoo/calm'
  },
  {
    snapshot: '/images/ikatoo.png',
    description: {
      title: 'iKatoo - Site pessoal',
      subTitle: 'Last update: 2022 - 08',
      content: 'Personal Web Site.'
    },
    githubLink: 'https://github.com/mckatoo/ikatoo-frontend'
  }
]
