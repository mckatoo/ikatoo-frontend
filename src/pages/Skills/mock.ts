import { SkillsProps } from '.'

export const mockSkillsPageData: SkillsProps = {
  title: 'Habilidades e Experiências',
  description: `<p>Trabalhei como programador em VBA usando o MS Access em 2003, logo após formado no curso técnico do SENAC. Não dei muita sorte pois a empresa fechou 3 meses após eu entrar.</p>
  <p>Precisando de emprego mudei de área, fui trabalhar em uma grande indústria farmacêutica da cidade e lá fiquei durante 7 anos trabalhando como operador de máquinas na produção de medicamentos injetáveis, mas sempre com um pézinho na programação. Lá eu criei um programa em Delphi 7 com banco em MS Access para controle de produção de medicamentos liofilizados que utilizávamos para melhorar o fluxo de produção do setor e diminuir o tempo nos processos de fabricação.</p>
  <p>Sempre estudando banco de dados e programação, em 2013 surgiu uma oportunidade na faculdade onde trabalhei e estudei até julho de 2021.</p>
  <p>Lá atuei principalmente na manutenção e implementação de rede de computadores com 3 laboratórios de informática, departamentos administrativos e outros departamentos academicos. Implementei uma revista eletronica "Consciesi" e desenvolvi pequenas ferramentas para auxiliar o sistema administrativo e acadêmico.</p>
  <p>Atualmente estou desempregado procurando novas oportunidades.</p>
  <p>Visite meu perfil no <a target="_blank" href="https://linkedin.com/in/mckatoo">LinkedIn</a> para mais detalhes ou entre em <a href="/contact">contato</a> comigo.`,
  skills: [
    { skillTitle: 'Back-end', rankPercent: 40 },
    { skillTitle: 'Modelagem de dados', rankPercent: 30 },
    { skillTitle: 'Front-end', rankPercent: 20 },
    { skillTitle: 'CI/CD', rankPercent: 10 }
  ],
  lastJobs: [
    {
      jobTitle: 'Calm Organizador de Criptomoedas',
      jobDescription: 'Projeto pessoal para estudo.',
      yearMonthStart: '2022 - 03'
    },
    {
      jobTitle: 'Uniesi - Centro Universitário de Itapira',
      jobDescription:
        'Responsável pela infraestrutura local e suporte dos serviços dispostos pela mantenedora UNIP.',
      yearMonthStart: '2013 - 06',
      yearMonthEnd: '2021 - 07'
    },
    {
      jobTitle: 'Itacom Veículos',
      jobDescription:
        'Desenvolvimento do módulo venda de veículos do sistema de gerenciamento geral.',
      yearMonthStart: '2003 - 10',
      yearMonthEnd: '2003 - 12'
    }
  ]
}
