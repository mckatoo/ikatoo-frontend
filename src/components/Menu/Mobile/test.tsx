import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { describe, it } from 'vitest'
import MobileMenu from '.'

import { mockedMenu } from '../mock'

describe('Menu Component', () => {
  it('should render menu on desktop', () => {
    const { getByText, getByLabelText, container } = render(
      <MemoryRouter initialEntries={[`/`]}>
        <MobileMenu links={mockedMenu.links} social={mockedMenu.social} />
      </MemoryRouter>
    )

    expect(getByLabelText(/mobile menu/i)).toBeInTheDocument()
    expect(getByLabelText(/logotipo/i)).toBeInTheDocument()
    expect(getByText(/sobre/i)).toHaveClass('text-mck_aqua')
    expect(getByText(/habilidades/i)).toHaveClass('text-gray-500')
    expect(getByText(/habilidades/i)).not.toHaveClass('text-mck_aqua')
    expect(container).toMatchSnapshot()
  })
})
