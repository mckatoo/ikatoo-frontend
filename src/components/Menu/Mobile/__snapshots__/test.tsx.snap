// Vitest Snapshot v1

exports[`Menu Component > should render menu on desktop 1`] = `
.c0 svg {
  margin: auto;
}

.c1 a,
.c1 span {
  display: block;
  padding: 1rem 2.5rem;
  width: 100%;
  cursor: pointer;
}

.c2 {
  --tw-bg-opacity: 1;
  background-color: rgb(8 253 216 / var(--tw-bg-opacity));
}

.c2 a {
  --tw-text-opacity: 1;
  color: rgb(0 3 44 / var(--tw-text-opacity));
}

<div>
  <menu
    aria-label="Mobile Menu"
    class="fixed text-center text-gray-500 h-16"
  >
    <div
      class="bg-slate-800 flex flex-row justify-between items-center w-screen"
    >
      <div
        class="w-12 h-12 my-2 mx-4 cursor-pointer"
      >
        <div
          aria-label="Logotipo"
          class="c0 text-gray-500 w-fit h-fit flex flex-col items-center text-center"
        >
          <a
            href="/"
          >
            <svg
              class="w-full"
              fill="none"
              viewBox="0 0 152 152"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="76"
                cy="76"
                r="74"
                stroke="url(#a)"
                stroke-width="4"
              />
              <path
                d="M81.5 62.5h39a5 5 0 0 1 5 5v17a5 5 0 0 1-5 5H91.543"
                stroke="#EF5321"
                stroke-width="3"
              />
              <path
                clip-rule="evenodd"
                d="M34.5 41a6.5 6.5 0 0 0-6.5 6.5v58a6.5 6.5 0 0 0 6.5 6.5h71a6.5 6.5 0 0 0 6.5-6.5v-15h-3v15a3.5 3.5 0 0 1-3.5 3.5h-71a3.5 3.5 0 0 1-3.5-3.5v-58a3.5 3.5 0 0 1 3.5-3.5h71a3.5 3.5 0 0 1 3.5 3.5v14h3v-14a6.5 6.5 0 0 0-6.5-6.5h-71Z"
                fill="#EF5321"
                fill-rule="evenodd"
              />
              <circle
                cx="75"
                cy="63"
                r="6"
                stroke="#EF5321"
                stroke-width="3"
              />
              <circle
                cx="54"
                cy="76"
                r="6"
                stroke="#EF5321"
                stroke-width="3"
              />
              <path
                d="M60 75.5h45"
                stroke="#EF5321"
                stroke-width="3"
              />
              <circle
                cx="111"
                cy="76"
                r="6"
                stroke="#EF5321"
                stroke-width="3"
              />
              <circle
                cx="85"
                cy="90"
                r="6"
                stroke="#EF5321"
                stroke-width="3"
              />
              <defs>
                <lineargradient
                  gradientUnits="userSpaceOnUse"
                  id="a"
                  x1="152"
                  x2="4.102"
                  y1="0"
                  y2="155.891"
                >
                  <stop
                    offset=".132"
                    stop-color="#212121"
                  />
                  <stop
                    offset="1"
                    stop-color="#3A3A41"
                  />
                </lineargradient>
              </defs>
            </svg>
          </a>
        </div>
      </div>
      <div
        class="w-12 h-12 my-2 mx-4 cursor-pointer"
      >
        <svg
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </div>
    <div
      class="h-[calc(100vh-4rem)] w-screen bg-slate-800 flex flex-col transition-opacity duration-300 opacity-0 pointer-events-none z-0"
    >
      <nav
        class="text-lg h-full flex flex-col justify-center -mt-4"
      >
        <ul
          class="list-reset border-slate-700 border-y-2 divide-y divide-slate-700 font-semibold"
        >
          <li
            class="c1"
          >
            <a
              class="text-mck_aqua"
              href="/about"
            >
              Sobre
            </a>
          </li>
          <li
            class="c1"
          >
            <a
              class="text-gray-500"
              href="/skills"
            >
              Habilidades
            </a>
          </li>
          <li
            class="c1"
          >
            <a
              class="text-gray-500"
              href="/projects"
            >
              Projetos
            </a>
          </li>
          <li
            class="c1"
          >
            <a
              class="text-gray-500"
              href="/contact"
            >
              Contato
            </a>
          </li>
          <li
            class="c1 c2"
          >
            <a
              href="/login"
            >
              Entrar
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="flex flex-row justify-around py-2 absolute bottom-0 w-full"
      >
        <a
          class="w-6 text-inherit"
          href="https://github.com/mckatoo"
          rel="noreferrer"
          target="_blank"
        />
        <a
          class="w-6 text-inherit"
          href="https://youtube.com/mckatoo"
          rel="noreferrer"
          target="_blank"
        />
        <a
          class="w-6 text-inherit"
          href="https://linkedin.com/in/mckatoo"
          rel="noreferrer"
          target="_blank"
        />
      </div>
    </div>
  </menu>
</div>
`
